### ISSUE TITLE
Roll promises are being swallowed by add promises

### ISSUE THREAD

**Issue by frankieali** [3d-dice/dice-box#17]
The recently updated `roll` and `add` methods now return a promise, but not all promises are being resolved correctly. I'm seeing issues when trying to add dice on a `setInterval` timer. This needs some rework

**Comment by frankieali** [comment]
Also easily demonstrable with
```javascript
Box.roll('3d6').then(()=>console.log('first'))
Box.add('3d6').then(()=>console.log('second'))
```

Only 'second' promise callback is triggered.

**Comment by frankieali** [comment]
Fixed in version 0.6. `Roll` and `Add` both return proper promises and so do `Reroll` and `Remove`. I normalized all the return data to be formatted the same. Readme documentation has been updated to explain that more. All four methods can now accept arrays of notations as well, but `roll` and `add` are still the only two that accept string notations.
